tests/subset.rs

```diff
@@ -1,39 +1,54 @@
 #![deny(unused_must_use)]
-//! Integration tests of the Subset protocol.
-
-mod network;
+pub mod net;
 
 use std::collections::{BTreeMap, BTreeSet};
 use std::iter::once;
 use std::sync::Arc;
 
 use hbbft::subset::{Subset, SubsetOutput};
-use hbbft::NetworkInfo;
+use hbbft::ConsensusProtocol;
 
-use crate::network::{Adversary, MessageScheduler, NodeId, SilentAdversary, TestNetwork, TestNode};
+use crate::net::adversary::{Adversary, NodeOrderAdversary, ReorderingAdversary};
+use crate::net::{NetBuilder, NewNodeInfo, VirtualNet};
 
+type NodeId = u16;
 type ProposedValue = Vec<u8>;
 
-fn test_subset<A: Adversary<Subset<NodeId, u8>>>(
-    mut network: TestNetwork<A, Subset<NodeId, u8>>,
+fn test_subset<A>(
+    mut net: VirtualNet<Subset<NodeId, u8>, A>,
     inputs: &BTreeMap<NodeId, ProposedValue>,
-) {
-    let ids: Vec<NodeId> = network.nodes.keys().cloned().collect();
+) where
+    A: Adversary<Subset<NodeId, u8>>,
+{
+    let ids: Vec<NodeId> = net.nodes().map(|node| *node.id()).collect();
+
+    let mut rng = rand::thread_rng();
```

Comment:
Please use `proptest::TestRng` to make tests reproducible given a seed. See `net_dynamic_hb.rs` and `binary_agreement_mitm.rs` for the relevant `proptest` macros that allow that.

Reply:
Switched to proptest and TestRng